### ISSUE TITLE
arangodb-java-driver async 6.5.0 is blocking application exit

### ISSUE THREAD

**Issue by cerveada** [arangodb/arangodb-java-driver#333]
Hello,
there seems to be a bug in Arangodb Java driver 6.5.0 (async API) that keeps application running even though it should exit. This is a problem especially for command line applications.

Probably some threads are not terminated properly during `arangoDB.shutdown();`

I created small project that replicates the issue when you run it: [arangodb-bug.tar.gz](https://github.com/arangodb/arangodb-java-driver/files/4122248/arangodb-bug.tar.gz)

The code is basically taken from async examples here.
```java
package foo;

import com.arangodb.async.ArangoCollectionAsync;
import com.arangodb.async.ArangoDBAsync;
import com.arangodb.async.ArangoDatabaseAsync;

public class Main 
{

    protected static final String COLLECTION_NAME = "json_example_collection";
    private static final String DB_NAME = "json_example_db";
    protected static ArangoDatabaseAsync db;
    protected static ArangoCollectionAsync collection;
    private static ArangoDBAsync arangoDB;

    public static void main(String[] args) throws Exception
    {
        System.out.println( "Hello Arango!" );

        arangoDB = new ArangoDBAsync.Builder().build();
        if (arangoDB.db(DB_NAME).exists().get()) {
            arangoDB.db(DB_NAME).drop().get();
        }
        arangoDB.createDatabase(DB_NAME).get();
        db = arangoDB.db(DB_NAME);
        db.createCollection(COLLECTION_NAME).get();
        collection = db.collection(COLLECTION_NAME);

        // tear down
        db.drop().get();
        arangoDB.shutdown();

        System.out.println( "END" );
    }
}
```

**Comment by rashtao** [comment]
fixed here: https://github.com/arangodb/arangodb-java-driver/commit/2189e809bb647bba4d17ee9cfea57edc97072c7a

Thanks for reporting

**Comment by rashtao** [comment]
Fix released in version 6.6.0
https://github.com/arangodb/arangodb-java-driver/releases/tag/v6.6.0
